### DISCUSSION TITLE
How to add Request Payload in Cypress (JS)

### DISCUSSION THREAD

**Discussion by sait-yamdu** [cypress-io/cypress#28508] [Questions and Help] [answered]
I know how to add query param in Cypress like that;
```
cy.request({
  url: "link",
  qs: {
    "projectId": "3494"
 }
})
```

But I don't know how to add this Request Payload ["288"] to the cy.request section. Please check the screenshot. Do you guys have any idea?
![image](https://github.com/cypress-io/cypress/assets/143401052/6414ddbf-ccf3-4707-885e-526b1bb7f5d3)

**Comment by portyu9** [comment, marked as answer]
Hi @sait-yamdu 

`qs` is for URL query parameters. To send:

```json
["288"]
```

as the request payload, use the `body` option.

For example:

```js
cy.request({
  method: 'POST',
  url: 'link',
  qs: {
    projectId: '3494',
  },
  body: ['288'],
})
```

That gives you a request conceptually like:

```text
POST link?projectId=3494
```

with:

```json
["288"]
```

in the request body.

If your API specifically requires the JSON content type, you can also provide:

```js
headers: {
  'Content-Type': 'application/json',
}
```

The important distinction is:

- `qs` → query-string parameters.
- `body` → request payload.

Also specify `method: 'POST'` (or the actual method your API requires), because `cy.request()` defaults to `GET` when no method is supplied.

Please mark this answer as accepted if it helped, thank you.
